soroban-sdk/src/env.rs

```diff
@@ -495,19 +497,18 @@ impl Env {
     }
 
     fn register_contract_with_source(&self, source: xdr::ScContractCode) -> BytesN<32> {
-        use crate::testutils::random_account_id;
-        use crate::testutils::random_bytes_array;
         let prev_source_account = if let Ok(prev_acc) = self.env_impl.source_account() {
             Some(prev_acc)
         } else {
             None
         };
-        self.env_impl.set_source_account(random_account_id());
+        self.env_impl
+            .set_source_account(AccountId::random(self).try_into().unwrap());
 
         let contract_id: BytesN<32> = self
             .env_impl
             .invoke_function(xdr::HostFunction::CreateContract(xdr::CreateContractArgs {
-                contract_id: xdr::ContractId::SourceAccount(xdr::Uint256(random_bytes_array())),
+                contract_id: xdr::ContractId::SourceAccount(xdr::Uint256(random())),
```

Comment:
nit: I think unlike the other changed functions (which are both more concise and clear) this one is ambiguous. Since it doesn't belong to any 'namespace'/struct, it's hard to tell what exactly does it generate, so I would probably go with something more specific (like the previous name or at least `random_array` or something like that).

Reply:
I'm in two minds about whether we should even expose this function at this stage. I think I'd rather folks import a RNG that they choose to use in their applications, so I'll make `random()` `pub(crate)` so it isn't part of the public API.

Reply:
I guess we have random on types now, so exposing this here is not that big of a deal.

Reply:
Ok, that's probably fine then, given that the function isn't public.

Reply:
I'll remove it, then lets see how much a need continues to arise for it. Most uses of random in contract tests appear to be people generating contract IDs, but that's all going away.

Reply:
Removed in d11aad01.